model/host/host.go

```diff
@@ -1742,3 +1758,74 @@ func (h *Host) SetInstanceType(instanceType string) error {
 	h.InstanceType = instanceType
 	return nil
 }
+
+// CountHostsWithNoExpirationByUser returns a count of all hosts associated
+// with a given users that are considered up and should never expire.
+func CountHostsWithNoExpirationByUser(user string) (int, error) {
+	query := db.Query(bson.M{
+		UserKey:         user,
```

Comment:
Should this actually check the StartedBy field? I forget how the data model is set, but we have an index on StartedBy but not User

Reply:
Yes! Good point, traced through at StartedBy is what we need.